pysatModels/models/pysat_dineof.py

```diff
@@ -0,0 +1,237 @@
+# -*- coding: utf-8 -*-
+"""
+Supports exported model data from pysatDINEOF, a Python package
+that interfaces with a version of Data Interpolation Empirical
+Orthogonal Functions (DINEOFs). This module couples into
+the systematic export pysatDINEOF format and thus should support
+all exports from the package.
+
+Given the generality of the support, each model series is identified
+using the `tag` keyword.
+
+DINEOFs are a purely data based method that can analyze a data-set, with
+data gaps, and extract a series of basis functions that optimally reproduce
+the input data. The quality of the reconstruction is primarily determined
+by the quantity and quality of the input data.
+
+http://modb.oce.ulg.ac.be/mediawiki/index.php/DINEOF
+
+References
+----------
+J.-M. Beckers and M. Rixen. EOF calculations and data filling from
+incomplete oceanographic data sets. Journal of Atmospheric and
+Oceanic Technology, 20(12):1839-­1856, 2003.
+
+Parameters
+----------
+platform : string
+    pysat
+name : string
+    dineof
+sat_id : string
+    ['']
+tag : string
+    [*]
+
+Note
+----
+::
+
+    Specific tags are not listed here as this method is intended to support
+    all pysatDINEOF export models. Place the desired model (daily files)
+    at '{pysat_data_dir}/pysat/dineof/{tag}'. It is presumed the default
+    naming scheme of 'dineof_model_{year:04d}-{month:02d}-{day:02d}.nc'
+    has been retained. Use the file_format option for custom filenames.
+        imodule = pysatModels.instruments.pysat_dineof
+        model = pysat.Instrument(inst_module=imodule, tag=tag)
+
+Warnings
+--------
+
+
+Authors
+-------
+
+
+"""
+
+# python 2/3 comptability
+from __future__ import print_function
+from __future__ import absolute_import
+
+import logging
+import pysat
+
+logger = logging.getLogger(__name__)
+
+# the platform and name strings associated with this instrument
+# need to be defined at the top level
+platform = 'pysat'
```

Comment:
Should the platform be pysat or UTD or something else?  I think this may be another instance where you went a bit overboard naming things pysat.  What's the main thing that separates this DINEOF implementation from others?

Reply:
Probably something else. I named it before I wrote it and pysat is mainly on the periphery. Main difference, Python interface to DINEOFs, increased bit-depth to support larger arrays, added support for multiprocessing,  plus I added support for line-of-sight vectors.

Reply:
What about renaming the package 'pyDINEOF' and following the 'sami2py' naming conventions?

Reply:
Internally in the package everything is pydineof. I think I forgot that is where I'm going, not where I was.